applications/tari_mining_node/src/stratum/controller.rs

```diff
@@ -108,13 +108,11 @@ impl Controller {
     }
 
     fn send_message(&mut self, message: &str) -> Result<(), Error> {
-        if self.stream.is_none() {
-            return Err(Error::Connection(String::from("No server connection")));
-        }
+        let stream = self.stream()?;
         debug!(target: LOG_TARGET_FILE, "sending request: {}", message);
-        let _ = self.stream.as_mut().unwrap().write(message.as_bytes());
```

Comment:
Why did we ignore these errors?

Reply:
:man_shrugging: What you've done looks correct

Reply:
Wow, this change helped to find a potential issue. It could miss some data on writing:

```
error: written amount is not handled. Use `Write::write_all` instead
   --> applications/tari_mining_node/src/stratum/controller.rs:113:9
    |
113 |         stream.write(message.as_bytes())?;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[deny(clippy::unused_io_amount)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_io_amount
```

Reply:
Good catch - write will return an error if the socket has to wait to write, previous code would have written junk in some cases.